### PR TITLE
Support rendering URLs by GUID

### PR DESCRIPTION
**Pull request by deviantintegral**
I already updated the CHANGELOG, see it for details!

### CODE REVIEW

File: src/Service/Player/Url.php [hunk 1]
```diff
@@ -123,10 +138,19 @@ public function __toString()
...
-    public function setAutoplay(bool $autoPlay)
+    public function withAutoplay(bool $autoPlay): self
     {
-        $this->autoPlay = $autoPlay;
+        if ($this->autoPlay == $autoPlay) {
+            return $this;
+        }
+
+        $url = clone $this;
+        $url->autoPlay = $autoPlay;
```
Comment (deviantintegral) [inline src/Service/Player/Url.php:158]: Apparently you can modify private properties when an object clones itself. 🍌 !

http://php.net/manual/en/language.oop5.cloning.php#122822

Comment (m4olivei) [reply src/Service/Player/Url.php:158]: Huh, look at that.

File: src/Service/Player/Url.php [hunk 2]
```diff
@@ -65,15 +68,22 @@ class Url implements ToUriInterface
...
      * Url constructor.
      *
      * @param PublicIdentifierInterface $account The account the player is owned by.
      * @param PublicIdentifierInterface $player  The player to play $media with.
-     * @param PublicIdentifierInterface $media   The media to play.
+     * @param ObjectInterface           $media   The media to play.
      */
-    public function __construct(PublicIdentifierInterface $account, PublicIdentifierInterface $player, PublicIdentifierInterface $media)
-    {
+    public function __construct(PublicIdentifierInterface $account, PublicIdentifierInterface $player, ObjectInterface $media
```
Comment (m4olivei) [inline src/Service/Player/Url.php]: Is this worth a note in the CHANGELOG? If people had been passing in `\Lullabot\Mpx\DataService\PublicIdentifierInterface` objects, that wouldn't fly anymore and would throw an error right?

**Review by m4olivei — APPROVED** [review]
Nice change. I wondered if we were eventually going to need to support the guid form of the URL. I like the immutability of the object, seems like a good fit for this class.

Just one question left, feel free to address and merge.

**Pull request merged**